Client/Frontend/Browser/Favorites/FavoritesViewController.swift

```diff
@@ -170,85 +205,309 @@ extension FavoritesViewController: KeyboardHelperDelegate {
+                            if fetchedObjects.count < totalCount {
+                                header.showButton.alpha = 1.0
+                                header.hideClearButton.alpha = 1.0
+                            } else if fetchedObjects.count == totalCount {
+                                header.showButton.alpha = 0.0
+                                header.hideClearButton.alpha = 1.0
+                            } else {
+                                header.showButton.alpha = 0.0
+                                header.hideClearButton.alpha = 0.0
+                            }
+                        } else {
+                            header.showButton.alpha = 0.0
+                            header.hideClearButton.alpha = 0.0
+                        }
+                    } else {
+                        header.showButton.alpha = 1.0
+                        header.hideClearButton.alpha = 1.0
+                    }
+                    return header
+                }
+            }
+            
         }
         return UICollectionReusableView()
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        // swiftlint:disable:next force_cast
-        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: FavoriteCell.identifier, for: indexPath) as! FavoriteCell
-        let fav = frc.object(at: IndexPath(item: indexPath.item, section: 0))
-        cell.textLabel.text = fav.displayTitle ?? fav.url
-        if let url = fav.url?.asURL {
-            // All favorites should have domain's, but it was noticed at one
-            // point that this wasn't the case, so for future bug-tracking
-            // assert if its not found.
-            assert(fav.domain != nil, "Domain should exist for all favorites")
-            // The domain for the favorite is required for pulling cached
-            // favicon info. Since all favorites should have persisted
-            // Domain's, we leave `persistent` as true
-            let domain = fav.domain ?? Domain.getOrCreate(forUrl: url, persistent: true)
-            cell.imageView.loadFavicon(siteURL: url, domain: domain, monogramFallbackCharacter: fav.title?.first)
-        }
-        cell.accessibilityLabel = cell.textLabel.text
-        return cell
+        guard let section = adjustedSection(indexPath.section) else {
+            assertionFailure("Invalid Section")
+            return UICollectionViewCell()
+        }
+        
+        switch section {
+        case .pasteboard:
+            assertionFailure("Pasteboard section should have no items")
+            return UICollectionViewCell()
+        case .favorites:
+            // swiftlint:disable:next force_cast
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: FavoriteCell.identifier, for: indexPath) as! FavoriteCell
```

Comment:
That would require rewriting the cell. This logic is from the previous code, it shows up as a change because I added the `adjustedSection` right before it.

Reply:
`CollectionViewReusable` still force unwraps:
```swift
// swiftlint:disable force_cast
  public func dequeueReusableCell<T: UICollectionViewCell & CollectionViewReusable>(for indexPath: IndexPath) -> T {
    return dequeueReusableCell(withReuseIdentifier: T.identifier, for: indexPath) as! T
  }
  // swiftlint:enable force_cast
  ```
  
  However, I have changed the code to use it.